### DISCUSSION TITLE
Get list of releases of my collection for each folder

### DISCUSSION THREAD

**Discussion by aurel1char** [joalla/discogs_client#64] [Support] [answered]
Hello,
Thanks for your time in this (great) project to maintain this discogs_client.
I am not expert in Python, but I could connect with the API ( d.identity() recognizes me, and i coud get the list of folders from my collection).
But i just could not get the list of releases included in my collection (bonus: for each collection folder, with release_id).
Could you tell me the corresponding code so that i understand how to request the base?
Thank you very much!

**Comment by JOJ0** [comment, marked as answer]
HI @aurel1char, you need to cycle through the collection folder's releases iterable. Something like this:
```
for item in collection_folders[0].releases:
    print(item)
    print(dir(item))
```

ID 0 is a special folder ID, it's got all the releases in your collection. ID's 1, 2, 3, ... are your particular folders.

HTH

**Reply by aurel1char** [reply]
Thank you very much, i'll try this.

**Comment by JOJ0** [comment]
If you find the time it would be wonderful if you submit a little chapter about that topic to our documentation. Actually we don't have anything about the basics of fetching collection data written down yet. A chapter like this would be nice: https://python3-discogs-client.readthedocs.io/en/latest/listing.html

You could write a simple md file similar to this one: https://github.com/joalla/discogs_client/blob/master/docs/source/listing.md
Doesn't have to be all the details. Some info about instantiating the collection, getting the folders and iterating through to get all the releases would be a great start.

Just saying...if you find the time...and know some basic markdown... It would help the project and most of all other users.

**Comment by JOJ0** [comment]
Here is an example of how to dig further down in the tree and e.g get the ID of an release: https://github.com/JOJ0/discodos/blob/b03a6ba8f95cee5f8e9362010e7ed46ab587f01f/discodos/model_collection.py#L316

Find out what attributes are available by using the dir() function as described in my first reply.

**Comment by prcutler** [comment]
Hi,

Sorry to resurrect an old thread, but I'd be willing to help out with some of the developer documentation examples.

First, thanks for maintaining the library and sharing the code examples above.   I have a lot of experience writing documentation for open source projects, but focused on users, not usually developers.  I sometimes find I learn best when I write down what I've learned and what a better way than to write some docs.

If you're willing to answer some basic questions  here in discussions as I get stuck building my app, I'll be more than happy to contribute a page like you use an example above.

Thanks!

Paul

**Reply by AnssiAhola** [reply]
Hi, Paul!

Thanks for the offer, more and better documentation wouldn't hurt 😅 
I'd be happy to answer any questions and I'm sure @alifhughes and @JOJ0 are as well.

btw. What kind of app are you building?

Anyway, if the questions are related to this problem spesifically you can ask them here, otherwise you can start a new discussion 🙂 

Cheers!
Anssi
